gnovm/pkg/gnolang/values.go

```diff
@@ -1458,6 +1460,18 @@ func (tv *TypedValue) AssertNonNegative(msg string) {
 	}
 }
 
+// randNaN64 returns an uint64 representation of NaN with a random payload.
+func randNaN64() uint64 {
+	const uvnan64 = 0x7FF8000000000001      // math.Float64bits(math.NaN())
+	return rand.Uint64()&^uvnan64 | uvnan64 //nolint:gosec
+}
+
+// randNaN32 returns an uint32 representation of NaN with a random payload.
+func randNaN32() uint32 {
+	const uvnan32 = 0x7FC00000              // math.Float32bits(float32(math.NaN()))
+	return rand.Uint32()&^uvnan32 | uvnan32 //nolint:gosec
```

Comment:
OK, so I'm not convinced by this approach in general, but especially for float32's.

We'd have `(32-9)` bits to play with for a random value, ie. 8388608. With this amount of values, finding a "birthday" isn't that absurd probability; for a 1% probability, you need 412 values; for 50%, 3411; for 99%, 8789.

https://www.bdayprob.com/

I remain on my proposal for a solution for this; ComputeMapKey should return a second field saying that the value is not comparable, and the computed key should not be stored in the vmap or used for comparisons.

Reply:
Good catch but i don't think this matters now. This probably should be  a  postlaunch task.

Reply:
I added logic to mitigate the possible collisions, and panic if the map is saturated with too many NaN values. No degradation of the general case.